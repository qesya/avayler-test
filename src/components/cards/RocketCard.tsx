import React from 'react'
import { Failure } from '../../../types/launch';
import ButtonOutline from '../buttons/Button'

interface IRocketCard {
  number: string;
  name: string;
  date_utc: string;
  cores: string;
  id: string;
  payloads: string;
  image: string;
  isSuccess: boolean;
  onClick: () => void;
  failure?: Array<Failure>;
}

const RocketCard: React.FC<IRocketCard> = ({ number, name, cores, date_utc, id, image, isSuccess, onClick, payloads, failure }) => {
  console.log('x', failure);
  return (
    <div className="flex flex-col md:flex-row justify-between my-10">
      <div className="flex flex-row">
        <h2 className="font-AbrilFatface text-5xl text-white mr-6">{number}</h2>
        <div className="flex flex-col">
          <h2 className="font-AbrilFatface text-5xl text-white mb-4">{name}</h2>
          <h4 className="font-Rajdhani text-white text-2xl">ID : {id}</h4>
          <h4 className="font-Rajdhani text-white text-2xl">Payloads : {payloads}</h4>
          <h4 className="font-Rajdhani text-white text-2xl">Date: {date_utc}</h4>
          <h4 className="font-Rajdhani text-white text-2xl">Cores : {cores ?? '-'}</h4>
          <h4 className="font-Rajdhani text-white text-2xl">Status : {isSuccess ? 'SUCCESS ðŸš€' : 'FAILURE ðŸš©'}</h4>
          {!isSuccess && <h4 className="font-Rajdhani text-white text-2xl">Reason of failure : {failure && failure.length > 0 ? '' : 'No Description'}</h4>}
          {failure?.map((fail, index) => {
            return (
              <div key={index} className="flex-1 px-8" data-testid="Test-Failure">
                <p className="font-Rajdhani text-white text-2xl">ðŸ”´ Reason : {fail.reason}</p>
                <p className="font-Rajdhani text-white text-2xl">ðŸ”´ Altitude: {fail.altitude ?? '-'}</p>
                <p className="font-Rajdhani text-white text-2xl">ðŸ”´ Time: {fail.time}</p>
              </div>
            )
          })}
        </div>
      </div>

      <img
        data-testid="Test-Image"
        src={image.length > 0 ? image : 'https://www.spacex.com/static/images/share.jpg'}
        className="w-24 h-24"
      />
    </div>
  )
}

export default RocketCard;
