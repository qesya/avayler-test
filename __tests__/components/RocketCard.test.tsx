import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import RocketCard from '../../src/components/cards/RocketCard';
import '@testing-library/jest-dom'

describe('Component:RocketCard.', () => {
  it('renders without crashing', async () => {
    const component = render(<RocketCard
      number="1"
      name="Crew-5"
      cores="633d9da635a71d1d9c66797b"
      date_utc="2017-01-14T17:54:00.000Z"
      id="62dd70d5202306255024d139"
      image="https://images2.imgbox.com/eb/d8/D1Yywp0w_o.png"
      isSuccess
      onClick={jest.fn()}
      payloads="631616a7ffc78f3b8567071a"
    />);
    expect(component).toMatchSnapshot();
  });

  it('it should render failure', async () => {
    render(<RocketCard
      number="1"
      name="Crew-5"
      cores="633d9da635a71d1d9c66797b"
      date_utc="2017-01-14T17:54:00.000Z"
      id="62dd70d5202306255024d139"
      image="https://images2.imgbox.com/eb/d8/D1Yywp0w_o.png"
      isSuccess={false}
      onClick={jest.fn()}
      payloads="631616a7ffc78f3b8567071a"
      failure={[{ reason: '', time: 0, altitude: '' }]}
    />);
    const failureComponent = screen.getByTestId('Test-Failure');
    expect(failureComponent).toBeInTheDocument();
  });

  it('it should render success', async () => {
    render(<RocketCard
      number="1"
      name="Crew-5"
      cores="633d9da635a71d1d9c66797b"
      date_utc="2017-01-14T17:54:00.000Z"
      id="62dd70d5202306255024d139"
      image="https://images2.imgbox.com/eb/d8/D1Yywp0w_o.png"
      isSuccess
      onClick={jest.fn()}
      payloads="631616a7ffc78f3b8567071a"
    />);
    const failureComponent = screen.queryByText(/ðŸ”´ Reason : /i);
    expect(failureComponent).not.toBeInTheDocument();
  });

  it('it should render default image', async () => {
    const expectedImage = "https://www.spacex.com/static/images/share.jpg";
    const { getByTestId } = render(<RocketCard
      number="1"
      name="Crew-5"
      cores="633d9da635a71d1d9c66797b"
      date_utc="2017-01-14T17:54:00.000Z"
      id="62dd70d5202306255024d139"
      image=""
      isSuccess
      onClick={jest.fn()}
      payloads="631616a7ffc78f3b8567071a"
    />);
    expect(getByTestId('Test-Image')).toHaveAttribute('src', expectedImage);
  });
});