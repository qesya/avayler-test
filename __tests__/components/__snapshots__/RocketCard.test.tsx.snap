// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:RocketCard. renders without crashing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col md:flex-row justify-between my-10"
      >
        <div
          class="flex flex-row"
        >
          <h2
            class="font-AbrilFatface text-5xl text-white mr-6"
          >
            1
          </h2>
          <div
            class="flex flex-col"
          >
            <h2
              class="font-AbrilFatface text-5xl text-white mb-4"
            >
              Crew-5
            </h2>
            <h4
              class="font-Rajdhani text-white text-2xl"
            >
              ID : 
              62dd70d5202306255024d139
            </h4>
            <h4
              class="font-Rajdhani text-white text-2xl"
            >
              Payloads : 
              631616a7ffc78f3b8567071a
            </h4>
            <h4
              class="font-Rajdhani text-white text-2xl"
            >
              Date: 
              2017-01-14T17:54:00.000Z
            </h4>
            <h4
              class="font-Rajdhani text-white text-2xl"
            >
              Cores : 
              633d9da635a71d1d9c66797b
            </h4>
            <h4
              class="font-Rajdhani text-white text-2xl"
            >
              Status : 
              SUCCESS ðŸš€
            </h4>
          </div>
        </div>
        <img
          class="w-24 h-24"
          data-testid="Test-Image"
          src="https://images2.imgbox.com/eb/d8/D1Yywp0w_o.png"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col md:flex-row justify-between my-10"
    >
      <div
        class="flex flex-row"
      >
        <h2
          class="font-AbrilFatface text-5xl text-white mr-6"
        >
          1
        </h2>
        <div
          class="flex flex-col"
        >
          <h2
            class="font-AbrilFatface text-5xl text-white mb-4"
          >
            Crew-5
          </h2>
          <h4
            class="font-Rajdhani text-white text-2xl"
          >
            ID : 
            62dd70d5202306255024d139
          </h4>
          <h4
            class="font-Rajdhani text-white text-2xl"
          >
            Payloads : 
            631616a7ffc78f3b8567071a
          </h4>
          <h4
            class="font-Rajdhani text-white text-2xl"
          >
            Date: 
            2017-01-14T17:54:00.000Z
          </h4>
          <h4
            class="font-Rajdhani text-white text-2xl"
          >
            Cores : 
            633d9da635a71d1d9c66797b
          </h4>
          <h4
            class="font-Rajdhani text-white text-2xl"
          >
            Status : 
            SUCCESS ðŸš€
          </h4>
        </div>
      </div>
      <img
        class="w-24 h-24"
        data-testid="Test-Image"
        src="https://images2.imgbox.com/eb/d8/D1Yywp0w_o.png"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
